src/main/java/chessgame/domain/Board.java

```diff
@@ -0,0 +1,114 @@
+package chessgame.domain;
+
+import chessgame.domain.piece.Knight;
+import chessgame.domain.piece.Pawn;
+import chessgame.domain.piece.Piece;
+import chessgame.domain.point.Point;
+
+import java.util.Collections;
+import java.util.Map;
+
+public class Board {
+    private final Map<Point, Piece> board;
+
+    public Board(Map<Point, Piece> board) {
+        this.board = board;
+    }
+
+    public Map<Point, Piece> getBoard() {
+        return Collections.unmodifiableMap(board);
+    }
+
+    public void move(Point source, Point target, Team turn) {
+        Piece piece = board.get(source);
+        checkSamePoint(source, target);
+        checkSource(source, turn);
+        boolean hasTarget = checkTarget(target, turn);
+        executeMove(source, target, piece, hasTarget);
+    }
+
+    private void executeMove(Point source, Point target, Piece piece, boolean hasTarget) {
+        if (piece.isMovable(source, target)) {
+            checkPawnMove(piece, hasTarget);
+            followPieceRoute(source, target, piece);
+            return;
+        } else if (piece instanceof Pawn) {
+            checkPawnAttack(source, target, piece, hasTarget);
+            return;
+        }
+        throw new IllegalArgumentException("움직일 수 없다.");
```

Comment:
```suggestion
        throw new IllegalArgumentException("움직일 수 없습니다.");
```

예외 메시지는 읽는이를 고려해야해요. 지금은 해당 메시지가 프로젝트의 고객에게 직접 노출되고 있죠. 그렇기에 일관된 어조로 수정해주시면 고객이 당황하지 않을거에요.

Reply:
생각하지 못했던 부분 이였던 것 같습니다!
일관성 있도록 수정하였습니다!

Reply:
네, 예외 메시지는 의미 있게 작성하는 것도 중요하지만 고객에게 보여지는 메시지일 경우에는 더 신경써야겠죠~?